glfs.c

```diff
@@ -430,58 +430,6 @@ static char* tcmu_get_path( struct tcmu_device *dev)
 	return config;
 }
 
-
-static bool glfs_check_config(const char *cfgstring, char **reason)
-{
```

Comment:
rbd's lack of implementation was more of a oversight I think. At the time of its implementation, we did not even know about the dbus API use and the person doing the implementation probably thought like you where it is a duplication of open (at least that is what I thought when I reviewed the code for merging). 

The person that did the initial implementation does not use gwcli. It is not required for use with rbd. gwcli also only comes into play on the initial setup. If the link to the rbd device is down during a restart then check_config would be needed to catch this.

> IMHO, for handlers whose fs calls are local, its not a bad idea to have config_check() as they ensure the config string at the first place.

I did not understand the comment about the first place and local handlers. For local handlers like file_* you can just do the equivalent of a check_config in targetcli/rtslib before making the create call like how you would in your rbd/gwcli example.

Agree with you on the cost comment. Here are my issues though:

1. The netlink interface to add device is broken. Runner cannot pass back add/create results to the kernel, so if the handler's open fails, you end up with a partial invalid device, and you do not know it until the initiator scans and sends IO.

2. It sounds like we were using check_config to validate the values that most commonly would cause failures and device addition time. Note: I thought it was supposed to be done in rtslib/targetcli, but I cannot seem to figure out where it is being called.

Ideally, we would just fix #1 and drop the check_config interface as it would then be a duplicate of open and it would work for all cases (initial setup and restart create time and for all apps).

If we are not going to fix it properly, then I think you would at least want to do this in a common place so it works for both the initial setup and restart case. To do that you need to do it in rtslib or runner.